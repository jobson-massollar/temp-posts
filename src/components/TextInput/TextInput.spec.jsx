import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { InputText, TextInput } from '.'

const props = {
    label: 'label',
    onChange: jest.fn()
}

describe('<TextInput />', () => {

    it('should render', () => {
        render(<TextInput {...props} />)

        expect(screen.getByText(props.label)).toBeInTheDocument()   // Check the input label
        expect(screen.getByRole('searchbox')).toBeInTheDocument()   // Check the input element
    })

    it('should handle keyboard', () => {
        render(<TextInput {...props} />)

        const input = screen.getByRole('searchbox')
        const value = 'abc'

        userEvent.type(input, value)

        expect(input.value).toBe(value)     // Check the input content
        expect(props.onChange).toHaveBeenCalledTimes(value.length)  // Check the handle calls
    })

    it('should match snapshot', () => {
        const { container } =  render(<TextInput {...props} />)

        expect(container.firstChild).toMatchSnapshot()
    })
})