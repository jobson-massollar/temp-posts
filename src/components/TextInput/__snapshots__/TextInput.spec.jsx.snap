// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<div
  class="input-container"
>
  label
   
  <input
    maxlength="50"
    placeholder="Type your search here"
    size="50"
    type="search"
  />
</div>
`;
